site/globalEntityControl/GlobalEntitySelection.ts

```diff
@@ -79,3 +80,47 @@ export function subscribeChartToGlobalEntitySelection(
 export function pageContainsGlobalEntityControl() {
     return document.querySelector("[data-global-entity-control]") !== null
 }
+
+type GlobalEntitySelectionQueryParams = {
+    country?: string
+}
+
+class GlobalEntitySelectionUrl implements ObservableUrl {
+    globalEntitySelection: GlobalEntitySelection
+
+    constructor(globalEntitySelection: GlobalEntitySelection) {
+        this.globalEntitySelection = globalEntitySelection
+    }
+
+    @computed get params(): GlobalEntitySelectionQueryParams {
+        const params: GlobalEntitySelectionQueryParams = {}
+        const entities = this.globalEntitySelection.selectedEntities
+        // Do not add 'country' param unless at least one country is selected
+        if (entities.length > 0) {
+            params.country = EntityUrlBuilder.entitiesToQueryParam(
+                entities.map(entity => entity.code)
+            )
+        }
+        return params
+    }
+
+    @computed get debounceMode(): boolean {
+        return false
+    }
+
+    populateFromQueryStr(queryStr?: string) {
+        if (queryStr === undefined) return
+        this.populateFromQueryParams(strToQueryParams(queryStr))
+    }
+
+    private populateFromQueryParams(params: QueryParams) {
+        if (params.country) {
+            const countryCodes = EntityUrlBuilder.queryParamToEntities(
+                params.country
+            )
+            this.globalEntitySelection.mode =
+                GlobalEntitySelectionModes.override
+            this.globalEntitySelection.selectByCountryCodes(countryCodes)
+        }
+    }
+}
```

Comment:
What's with these changes, do they have to do with this PR?

Reply:
Yes. I need to refactor Entity Selection for the Energy Explorer Single Country Multi Metric charts to work (and will help with other things as well). So trying to isolate all Selection code and find some patterns to simplify things.